python/ray/experimental/gpu_object_manager/gpu_object_manager.py

```diff
@@ -3,12 +3,30 @@
 import ray
 from ray._private.custom_types import TensorTransportEnum
 from ray._raylet import ObjectRef
-
+import ray.util.collective as collective
+from ray.util.collective.types import Backend
+from dataclasses import dataclass
 
 if TYPE_CHECKING:
     import torch
     from ray.experimental.gpu_object_manager.gpu_object_store import GPUObjectStore
 
+
+@dataclass
+class TensorMetadata:
+    """Metadata for tensors stored in the GPU object store.
+
+    Args:
+        tensor_basic_meta: A list of tuples, each containing the shape and dtype of a tensor.
+        nixl_serialized_descs: Serialized NIXL descriptors for NIXL transport.
+        nixl_agent_meta: The additional metadata of the remote NIXL agent.
+    """
+
+    tensor_basic_meta: List[Tuple["torch.Size", "torch.dtype"]]
+    nixl_serialized_descs: Optional[bytes] = None
+    nixl_agent_meta: Optional[bytes] = None
+
```

Comment:
Instead of always passing optional nixl metadata, can we subclass TensorTransportMetadata so that we abstract away which tensor transport is being used?

E.g., `__ray_get_tensor_meta__` could be changed to something like this:

```python
def __ray_get_tensor_meta__(self: "ray.actor.ActorHandle", obj_id: str, tensor_transport: TensorTransportEnum) -> TensorTransportMetadata
```

and it should return a subclass of TensorTransportMetadata that matches the given tensor_transport.

Then we can subclass with something like NixlTransportMetadata and CollectiveTransportMetadata, and CollectiveTransportMetadata just contains the src rank.